### PR TITLE
Converted viewAccountRecord to use lightning-record-form

### PR DESCRIPTION
**Pull request by dbreese**
Also showing an example on how to view and interact with drafts.

@khawkins @eric-romero @AdamLiechty @maliroteh-sf @sfdctaka @pvandyk

### CODE REVIEW

File: force-app/main/default/lwc/viewAccountRecord/__tests__/data/getRecord.json [hunk 1]
```diff
@@ -18,5 +18,19 @@
         "Type": {
             "value": "Record type"
         }
+    },
+
+    "drafts": {
```
Comment (dbreese) [inline force-app/main/default/lwc/viewAccountRecord/__tests__/data/getRecord.json:23]: For unit tests to demonstrate how to see previous/old values.

File: force-app/main/default/lwc/viewAccountRecord/viewAccountRecord.html [hunk 2]
```diff
@@ -1,41 +1,50 @@
...
-                    <c-offline-field-view label="Owner" record={record} name="Owner"></c-offline-field-view>
-                    <c-offline-field-view label="Industry" record={record} name="Industry"></c-offline-field-view>
-                    <c-offline-field-view label="Type" record={record} name="Type"></c-offline-field-view>
+                    <!-- use lightning-record-form to show the compact layout. here, we use 'view' as the mode which allows edits too -->
+                    <lightning-record-form record-id={recordId} object-api-name={objectApiName} layout-type="Compact"
+                        columns="1" mode="view">
+                    </lightning-record-form>
                 </div>
+
+                <!-- show any draft information for this record -->
```
Comment (dbreese) [inline force-app/main/default/lwc/viewAccountRecord/viewAccountRecord.html:33]: This gets rid of need to have offline-field-view custom component.

File: force-app/main/default/lwc/viewAccountRecord/__tests__/data/getRecord.json [hunk 3]
```diff
@@ -18,5 +18,19 @@
...
+        "serverValues" : {
+            "Name" : {
+                "value" : "Old Name",
+                "displyValue" : null
```
Comment (khawkins) [inline force-app/main/default/lwc/viewAccountRecord/__tests__/data/getRecord.json:27]: Is `displyValue` a typo?

**Review by khawkins — APPROVED** [review]

File: force-app/main/default/lwc/viewAccountRecord/__tests__/data/getRecord.json (hunk 3 above)
Comment (dbreese) [reply force-app/main/default/lwc/viewAccountRecord/__tests__/data/getRecord.json:27]: Eagle-eyes!! Good catch. It is actually a typo, but is more for demo purposes as it is not used by any of the code (I just wanted to have a more real-life example in the mock data file (even though I bunged it up with a typo!). I'll fix and push.

**Comment by agarwasg** [comment]
@AdamLiechty  @eric-romero ping  to check if moving to lightning-record-form will not impact of offlinability and LSDK compatibility.

**Comment by eric-romero** [comment]
@agarwasg the data loads offline, but looks like the hostname bug occurs with these changes (we've only seen this previously in the current edit LWC). the edit mode is a little clumsy as well and the SAVE didnt appear to work. We should consider using read-only for lightning-record-form for the `.view` LWC

Oct-18-2022 09-47-05

**Comment by dbreese** [comment]
Interesting, I have not seen that error and have tested quite a bit over last week, but I'm on 240 versions of LSDK and my org. What version of LSDK are you testing with and org version (242 or 240)?

Are you saying that I should change the LWC to use "readOnly" and you'll be ok with the changes?

**Comment by eric-romero** [comment]
Yes, I would be ok if we use readOnly for the 'view' component.

In your testing were you only testing offline-ability after _passively_ caching the LWC? Or did you try calling `prefetch(lwc)` first before viewing the LWC offline (without ever loading it after a fresh login)?

I ask because the latter case is what we are mostly interested in, and that is what is shown in the gif. As I mentioned the 'hostname' bug has been reported before so I'm not suggesting we try to fix it here or anything like that

**Comment by dbreese** [comment]
Ah, I am not sure how to even call prefetch() -- I'm just logging in and launching the LWC.

I'll update this to use read-only for the record-form.

**Pull request merged**